### PR TITLE
Anomaly Detection: Add Daily Season with Hourly Interval to HoltWinter

### PR DESCRIPTION
**Pull request by zeotuan**
Add Daily Season with Hourly Interval to HoltWinter: #296
Add constructor that directly take seriesPeriodicity

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by zeotuan** [comment]
[rdsharma26](https://github.com/rdsharma26) Hi Can you help review this PR? It would be great if HoltWinter support more seasonality and interval. Also having a custom seriesPeriodicity would allow others to construct their own seasonality without always having to make changes to source code

**Comment by rdsharma26** [comment]
@zeotuan Thanks for the PR! The change looks good. However, would it be possible to add a unit test that shows how a custom value for `seriesPeriodicity` can be used?

**Comment by zeotuan** [comment]
@rdsharma26 Thanks. Tests Added for Custom seriesPeriodicity and Hourly Interval/Daily Seasonality

File: src/test/scala/com/amazon/deequ/anomalydetection/seasonal/HoltWintersTest.scala [hunk 1]
```diff
@@ -207,6 +207,73 @@ class HoltWintersTest extends AnyWordSpec with Matchers {
...
+  "work on hourly data with daily seasonality" in {
+    // https://www.kaggle.com/datasets/fedesoriano/traffic-prediction-dataset
+    val hourlyTrafficData = Vector[Double](
+      15, 13, 10, 7, 9, 6, 9, 8, 11, 12, 15, 17, 16, 15, 16, 12, 12, 16, 17, 20, 17, 19, 20, 15,
+      14, 12, 14, 12, 12, 11, 13, 14, 12, 22, 32, 31, 35, 26, 34, 30, 27, 27, 24, 26, 29, 32, 30, 27,
+      21, 18, 19, 13, 11, 11, 11, 14, 15, 29, 33, 32, 32, 29, 27, 26, 28, 26, 25, 29, 26, 24, 25, 20,
+      18, 18, 13, 13, 10, 12, 13, 11, 13, 22, 26, 27, 31, 24, 23, 26, 26, 24, 23, 25, 26, 24, 26, 24,
+      19, 20, 18, 13, 13, 9, 12, 12, 15, 16, 23, 24, 25, 24, 26, 22, 20, 20, 22, 26, 22, 21, 21, 21,
+      16, 18, 19, 14, 12, 13, 14, 14, 13, 20, 22, 26, 26, 21, 23, 23, 19, 19, 20, 24, 18, 19, 16, 17,
+      16, 16, 10, 9, 8, 7, 9, 8, 12, 13, 17, 14, 14, 14, 14, 11, 15, 13, 12, 17, 18, 17, 16, 15, 13,
```
Comment (rdsharma26) [inline src/test/scala/com/amazon/deequ/anomalydetection/seasonal/HoltWintersTest.scala:220]: Can you remove the last `,` ? The build is failing as a result.

File: src/test/scala/com/amazon/deequ/anomalydetection/seasonal/HoltWintersTest.scala [hunk 2]
```diff
@@ -207,6 +207,73 @@ class HoltWintersTest extends AnyWordSpec with Matchers {
       anomalies should have size 3
     }
   }
+
+  "work on hourly data with daily seasonality" in {
```
Comment (rdsharma26) [inline src/test/scala/com/amazon/deequ/anomalydetection/seasonal/HoltWintersTest.scala:211]: Can you add a test for custom `seriesPeriodicity` as well? The two tests are testing the original constructor and not the new constructor.

File: src/test/scala/com/amazon/deequ/anomalydetection/seasonal/HoltWintersTest.scala (hunk 1 above)
Comment (zeotuan) [reply src/test/scala/com/amazon/deequ/anomalydetection/seasonal/HoltWintersTest.scala]: Fixed

**Comment by zeotuan** [comment]
I have addressed the comments.

**Comment by rdsharma26** [comment]
Thanks for addressing the comments. LGTM.

**Review by rdsharma26 — APPROVED** [review]

**Pull request merged**